examples/language/llama2/finetune.py

```diff
@@ -0,0 +1,212 @@
+    parser.add_argument('-p',
+                        '--plugin',
+                        type=str,
+                        default='torch_ddp',
+                        choices=['torch_ddp', 'torch_ddp_fp16', 'gemini', 'low_level_zero', 'hybrid_parallel'],
+                        help="plugin to use")
+
+    parser.add_argument('--model_path', type=str, help="path to load model.")
+    parser.add_argument('--output_path', type=str, default=None, help="path to save model.")
+    parser.add_argument('--target_f1', type=float, default=None, help="target f1 score. Raise exception if not reached")
+    parser.add_argument('--use_lazy_init', type=bool, default=False, help="for initiating lazy init context")
+    args = parser.parse_args()
+
+    # ==============================
+    # Launch Distributed Environment
+    # ==============================
+    colossalai.launch_from_torch(config={}, seed=42)
+    coordinator = DistCoordinator()
+
+    # local_batch_size = BATCH_SIZE // coordinator.world_size
+    lr = LEARNING_RATE * coordinator.world_size
+
+    # ==============================
+    # Instantiate Plugin and Booster
+    # ==============================
+    booster_kwargs = {}
+    if args.plugin == 'torch_ddp_fp16':
+        booster_kwargs['mixed_precision'] = 'fp16'
+    if args.plugin.startswith('torch_ddp'):
+        plugin = TorchDDPPlugin()
+    elif args.plugin == 'gemini':
+        plugin = GeminiPlugin(initial_scale=2**5)
+    elif args.plugin == 'low_level_zero':
+        plugin = LowLevelZeroPlugin(initial_scale=2**5)
+    elif args.plugin == 'hybrid_parallel':
+
+        # modify the param accordingly for finetuning test cases
+        plugin = HybridParallelPlugin(tp_size=4,
+                                      pp_size=1,
+                                      num_microbatches=None,
+                                      microbatch_size=1,
+                                      enable_jit_fused=False,
+                                      zero_stage=0,
+                                      precision='fp32',
+                                      initial_scale=1)
+
+    booster = Booster(plugin=plugin, **booster_kwargs)
+
+    # ==============================
+    # Prepare Dataloader
+    # ==============================
+    data_builder = DataBuilder(args.model_path,
+                               plugin,
+                               args.task,
+                               train_batch_size=BATCH_SIZE,
+                               eval_batch_size=BATCH_SIZE)
+    train_dataloader = data_builder.train_dataloader()
+
+    # ====================================
+    # Prepare model, optimizer
+    # ====================================
+
+    cfg = AutoConfig.from_pretrained(args.model_path)
+
+    if args.use_lazy_init:
+        args.use_lazy_init = False
+        warnings.warn("lazy init is not compatible with from_pretrained now")
```

Comment:
Lazy init can be used by initializing from config and load model after `boost`

Reply:
You shoud add comments here to demonstrate in which case (model and plugin) we should use lazy init.